src/test/java/wolox/training/controllers/UserControllerTest.java

```diff
@@ -0,0 +1,284 @@
+import wolox.training.exceptions.responses.BookAlreadyOwnException;
+import wolox.training.exceptions.responses.UsernameAlreadyTakenException;
+import wolox.training.models.Book;
+import wolox.training.models.User;
+import wolox.training.repositories.UserRepository;
+
+@WebMvcTest(UserController.class)
+class UserControllerTest {
+
+    @Autowired
+    private MockMvc mvc;
+
+    @Autowired
+    private ObjectMapper mapper;
+
+    @MockBean
+    private UserRepository userRepository;
+
+    @Test
+    void whenGetAllUsers_thenReturnJsonArray() throws Exception {
+        //Given
+        User userTest = new User(null,"Username","David", LocalDate.parse("2004-09-25"),new ArrayList<>());
+        List<User> allUsers = Arrays.asList(userTest);
+
+        given(userRepository.findAll()).willReturn(allUsers);
+
+        //When + Then
+        mvc.perform(MockMvcRequestBuilders.get("/api/users")
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$",hasSize(1)))
+                .andExpect(jsonPath("$[0].name", is(userTest.getName())))
+                .andExpect(jsonPath("$[0].birthdate", is(userTest.getBirthdate().toString())));
+        System.out.println("whenGetAllUsers Passed");
+    }
+
+    @Test
+    void whenGetAnUserByUsername_thenReturnJsonArray() throws Exception {
+        //Given
+        User userTest = new User(null,"Username","David", LocalDate.parse("2004-09-25"),new ArrayList<>());
+
+        given(userRepository.findByUsername(userTest.getUsername())).willReturn(java.util.Optional.of(userTest));
+
+        //When + Then
+        mvc.perform(MockMvcRequestBuilders.get("/api/users?username=Username")
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$",hasSize(1)))
+                .andExpect(jsonPath("$[0].username", is(userTest.getUsername())));
+        System.out.println("whenGetAUserByUsername Passed");
+    }
+
+    @Test
+    void whenGetAnUserById_thenReturnJson() throws Exception {
+        //Given
+        User userTest = new User(1L,"Username","David", LocalDate.parse("2004-09-25"),new ArrayList<>());
+
+        given(userRepository.findById(userTest.getId())).willReturn(java.util.Optional.of(userTest));
+
+        //When + Then
+        mvc.perform(MockMvcRequestBuilders.get("/api/users/1")
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$").exists())
+                .andExpect(jsonPath("$", isA(LinkedHashMap.class)))
+                .andExpect(jsonPath("$.*").exists())
+                .andExpect(jsonPath("$.*", hasSize(5)))
+                .andExpect(jsonPath("$.id", is(userTest.getId().intValue())))
+                .andExpect(jsonPath("$.name", is(userTest.getName())));
+        System.out.println("whenGetAnUserById Passed");
```

Comment:
This print is not necessary, as it is a test, the result is already shown by the console